### DISCUSSION TITLE
Audible Alarms/Warnings

### DISCUSSION THREAD

**Discussion by valschmidt** [foxglove/community#326] [Q&A]
For operational robots, in addition to seeing time series and other plots of data, it is essential to have audible alarms that allow one to look away from the plots and focus on other things momentarily without fear of missing an emergency condition. 

Consider, for example, the state of a bilge pump on a robotic boat, the engine coolant temperature or oil pressure from an engine, or the range to an obstacle. Operators need to be warned of these things, and while ROS Diagnostics and other methods provide a framework for creating warnings and errors messages, these text messages are not enough during real-time operations when there are 1000's of things vying for one's attention. 

Any implementation of audible alarms should have some basic features. A threshold for alarm, a time above the threshold for alarm, both a visual indication (plot background flashes?) and audible alarm, and of course the ability to silence or disable an alarm - all the basic stuff.

**Comment by jtbandes** [comment]
Tangentially related to foxglove/community#147 (feature request for panel that can play audio data).

I wonder if this feature would actually make sense as a feature of the "Indicator" panel sketched out in foxglove/studio#2003?
